bridges/otelzap/zap_test.go

```diff
@@ -0,0 +1,353 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+// Copyright (c) 2016-2017 Uber Technologies, Inc.
+package otelzap
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"go.opentelemetry.io/otel/log"
+	"go.opentelemetry.io/otel/log/embedded"
+)
+
+var (
+	testBodyString = "log message"
+	testSeverity   = log.SeverityInfo
+	entry          = zapcore.Entry{
+		Level:   zap.InfoLevel,
+		Message: testBodyString,
+	}
+)
+
+type spyLogger struct {
+	embedded.Logger
+	Context context.Context
+	Record  log.Record
+}
+
+func (l *spyLogger) Emit(ctx context.Context, r log.Record) {
+	l.Context = ctx
+	l.Record = r
+}
+
+func (l *spyLogger) Enabled(ctx context.Context, r log.Record) bool {
+	return true
+}
+
+func NewTestOtelLogger(l log.Logger) zapcore.Core {
+	return &Core{
+		logger: l,
+	}
+}
```

Comment:
This looks unnecessary.

Reply:
For now you could simply inline it

```go
logger := &Core{ logger: l }
```

However, it may be better to simply copy-paste: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/d15fe4f317e0376ff4a65a435b785a1c88e11b7c/bridges/otelslog/handler_test.go#L35-L76.

@MrAlias. Do you think it would be worth to extract this code to a separate package and module?